src/functionaltests/java/com/ericsson/gerrit/plugins/eiffel/linking/GerritMockTest.java

```diff
@@ -0,0 +1,164 @@
+package com.ericsson.gerrit.plugins.eiffel.linking;
+
+import static org.junit.Assert.assertEquals;
+import static org.powermock.api.mockito.PowerMockito.mock;
+
+import java.io.IOException;
+
+import org.junit.Test;
+
+import com.ericsson.gerrit.plugins.eiffel.git.CommitInformation;
+import com.google.gerrit.extensions.restapi.ResourceNotFoundException;
+
+public class GerritMockTest {
+
+    @Test
+    public void oneUser() {
+        GerritMock gerritMock = new GerritMock();
+        String branch = "branch";
+        String user1 = "user1";
+        gerritMock.createBranch(branch);
+        GitCommit base = gerritMock.getHead(branch);
+
+        String changeId = gerritMock.createNewChange(user1, branch);
+        GitCommit update = gerritMock.getCommit(changeId);
+        assertEquals(base.sha, update.parentSha);
+
+        GitCommit newPatchSet = gerritMock.createNewPatchSet(changeId);
+        assertEquals(base.sha, newPatchSet.parentSha);
+
+        GitCommit merged = gerritMock.submit(changeId);
+        assertEquals(base.sha, merged.parentSha);
+        assertEquals(merged.sha, gerritMock.getHead(branch).sha);
+    }
+
+    @Test
+    public void oneUserTwoBranches() {
+        GerritMock gerritMock = new GerritMock();
+        String branch1 = "branch1";
+        String branch2 = "branch2";
+        String user1 = "user1";
+        gerritMock.createBranch(branch1);
+        gerritMock.createBranch(branch2);
+        GitCommit base = gerritMock.getHead(branch1);
+        GitCommit base2 = gerritMock.getHead(branch2);
+
+        String changeId = gerritMock.createNewChange(user1, branch1);
+        assertEquals(changeId, gerritMock.getChangeId(user1, branch1));
```

Comment:
Each assert should contain a message explaining what it asserts, leave out expected values in the message since this will be displayed anyway when the assert fails.

Reply:
Fixed in 6874169